### PR TITLE
faq updates

### PR DESCRIPTION
**Pull request by sethladd**
Fixes https://github.com/flutter/flutter/issues/9658

### CODE REVIEW

**Comment by sethladd** [comment]
@Hixie @eseidelGoogle @abarth PTAL if you're interested. Thanks!

File: faq.md [hunk 1]
```diff
@@ -8,136 +8,313 @@ permalink: /faq/
...
 mobile apps because Flutter uses neither WebView nor the OEM widgets
 that shipped with the device. Instead, Flutter uses its own high-performance
 rendering engine to draw widgets.
 
-Flutter also offers developers a highly productive and fast development
-experience, fast runtime and engine performance, and beautifully designed
-widgets that make for beautiful apps.
+In addition, Flutter is different because it only has a thin layer of
+C/C++ code. Flutter implements most of its system (compositing, gestures,
+animation, framework, widgets, etc) in _Dart_ (a modern, terse, OOP
```
Comment (abarth) [inline faq.md]: Is OOP too obscure an acronym?

File: faq.md [hunk 2]
```diff
@@ -8,136 +8,313 @@ permalink: /faq/
...
+Yes! Flutter ships with a functional-reactive style framework, inspired by
+React. Flutter's framework is designed to be layered and customizable
+(and optional).
+Developers can choose to use only parts of the framework, or a different
+framework.
+
+### Does Flutter come with widgets?
+
+Yes! Flutter ships with a set of
+[high quality Material and Cupertino (iOS-styled) widgets][widgets], layouts,
```
Comment (abarth) [inline faq.md]: styled or style ?  In the API docs we usually say style

File: faq.md [hunk 3]
```diff
@@ -8,136 +8,313 @@ permalink: /faq/
...
+course, these widgets are only a starting point.
+Flutter is designed to make it easy to
+create your own widgets, or customize the existing widgets.
 
-## What technology is Flutter built with?
+### Does Flutter come with a testing framework?
+
+Yes, Flutter provides APIs for writing unit and integration
+tests. Learn more about
+[testing with Flutter](/testing/).
```
Comment (abarth) [inline faq.md]: Do we want to advertise our passion for testing and link to our coveralls page?

File: faq.md [hunk 4]
```diff
@@ -8,136 +8,313 @@ permalink: /faq/
...
+Flutter should be able to run most Dart code that does not import (transitively,
+or directly) dart:mirrors or dart:html.
+
+### How big is the Flutter engine?
+
+In November 2015, we measured the size of a minimal Flutter app, bundled as
+an APK, to be approximately 8MB. For this simple app that used Material Design
+widgets, the core engine is approximately 5MB, the framework + app code is
+approximately 400kb, necessary Java code is 330k, and there is approximately
+2.5MB of ICU data. We are working to get this smaller.
```
Comment (abarth) [inline faq.md]: "We are working to get this smaller." is this important to say?  It seems apologetic.

**Comment by abarth** [comment]
LGTM

File: faq.md (hunk 1 above)
Comment (sethladd) [inline faq.md]: fair enough. changed

File: faq.md (hunk 2 above)
Comment (sethladd) [inline faq.md]: fixed

File: faq.md (hunk 3 above)
Comment (sethladd) [inline faq.md]: Sure! What's the link?

File: faq.md (hunk 4 above)
Comment (sethladd) [inline faq.md]: Removed that line.

File: faq.md [hunk 5]
```diff
@@ -1,143 +1,325 @@
...
+to approachability for the majority of the system.
 
-## Why would I want to invest in learning Flutter?
+### Should I build my next production app with Flutter?
 
-Flutter is an easy way to use a single codebase to deliver beautiful mobile
-apps that run on both Android and iOS. Flutter gives developers quick
-edit/debug cycles for an enjoyable low-friction workflow.
+Flutter is still being developed and is not yet at
+1.0.
```
Comment (xster) [inline faq.md]: Technically I think react native isn't 1.0 yet either. Should we be less conservative in the language as well?

**Pull request merged**